scipy/stats/_multicomp.py

```diff
@@ -24,10 +27,167 @@
 class DunnettResult:
     statistic: np.ndarray
     pvalue: np.ndarray
+    _alternative: Literal['two-sided', 'less', 'greater']
+    _rho: np.ndarray
+    _df: int
+    _std: float
+    _mean_samples: np.ndarray
+    _mean_control: np.ndarray
+    _n_samples: np.ndarray
+    _n_control: int
+    _rng: SeedType
+    _ci: ConfidenceInterval | None = None
+    _ci_cl: DecimalNumber | None = None
+
+    def __str__(self):
+        # Note: `__str__` prints the confidence intervals from the most
+        # recent call to `confidence_interval`. If it has not been called,
+        # it will be called with the default CL of .95.
+        if self._ci is None:
+            self.confidence_interval(confidence_level=.95)
+        s = (
+            "Dunnett's test"
+            f" ({self._ci_cl*100:.1f}% Confidence Interval)\n"
+            "Comparison               Statistic  p-value  Lower CI  Upper CI\n"
+        )
+        for i in range(self.pvalue.shape[0]):
+            s += (f" (Sample {i} - Control) {self.statistic[i]:>10.3f}"
+                  f"{self.pvalue[i]:>10.3f}"
+                  f"{self._ci.low[i]:>10.3f}"
+                  f"{self._ci.high[i]:>10.3f}\n")
+
+        if self._alternative == 'less':
+            s += (
+                "\nOne-sided alternative (less): sample i's mean exceed "
+                "the control's mean by an amount at least Lower CI"
```

Comment:
I'm not sure whether I'm a fan of this. R prints something like it, right? My concern is that it purports to interpret what a confidence interval *is*, but it doesn't really give the detail that is needed for a correct understanding. You could ask me to get over it, or we could wordsmith this to death, or we could just remove it : )

If we keep it, I would prefer "the difference between the mean of sample `i` and the mean of the control is..." I think that the order of subtraction is unambiguous, but it is easier to parse a negative *difference* than a negative amount for "exceeds".

Reply:
His wording was fine, but you would then need to add a caveat like he does every time he makes statements like these: "The joint statement consisting of the ... conclusions has a confidence coefficient of 95%." Without this, it looks like a statement of 100% certainty. And I'm not sure if we want to write that caveat verbatim, so we would need to wordsmith that.